jenkins-master.groovy

```diff
@@ -57,43 +32,27 @@ pipeline {
 
         stage('Deploy API container') {
             steps {
-                withCredentials([
-                    string(credentialsId: 'docker-registry-azure', variable: 'DRpass'),
-                    string(credentialsId: "captainhook-master-facebook-verify-token", variable: "captainhook_facebook_verify_token" ),
-                    usernamePassword(
-                        credentialsId: "captainhook-master-rabbit-username-password",
-                        passwordVariable: "captainhook_rabbit_password",
-                        usernameVariable: "captainhook_rabbit_username"
+                withCredentials([file(credentialsId: 'jenkins-roihunter-master-kubeconfig', variable: 'kube_config')]) {
+                    kubernetesDeploy(
+                        configs: '**/kubernetes/captain-hook-master-deployment.yaml,**/kubernetes/captain-hook-master-service.yaml',
+                        dockerCredentials: [
+                             [credentialsId: 'docker-azure-credentials', url: 'http://roihunter.azurecr.io']
+                        ],
+                        kubeConfig: [
+                            path: "$kube_config"
+                        ],
+                        secretName: 'regsecret',
+                        ssh: [
+                            sshCredentialsId: '*',
+                            sshServer: ''
+                        ],
+                        textCredentials: [
+                            certificateAuthorityData: '',
+                            clientCertificateData: '',
+                            clientKeyData: '',
+                            serverUrl: 'https://'
+                        ]
                     )
-                ]) {
-                    script {
-                        def servers = params.APP_SERVERS.tokenize(',')
-
-                        for (item in servers) {
-                            sshagent(['5de2256c-107d-4e4a-a31e-2f33077619fe']) {
-                                sh """ssh -oStrictHostKeyChecking=no -t -t jenkins@${item} <<EOF
-                                    docker login roihunter.azurecr.io -u roihunter -p "$DRpass"
-                                    docker pull roihunter.azurecr.io/captain-hook/master
-                                    docker stop captain-hook-master; true
-                                    docker rm -v captain-hook-master; true
-                                    docker run --detach -p 8007:8005 \
```

Comment:
@daniilsvirin I noticed that host port was 8007, but I guess it is not necessary now?

Reply:
Yes, we use the same port like the app in the container.